cmd/porter/bundle.go

```diff
@@ -295,7 +295,9 @@ func buildBundlePublishCommand(p *porter.Porter) *cobra.Command {
 		Long:  "Publishes a bundle by pushing the invocation image and bundle to a registry.",
 		Example: `  porter bundle publish
   porter bundle publish --file myapp/porter.yaml
-  porter bundle publish --archive /tmp/mybuns.tgz --tag myrepo/my-buns:0.1.0
+  porter bundle publish --archive /tmp/mybuns.tgz --reference myrepo/my-buns:0.1.0
+  porter bundle publish --tag latest
```

Comment:
It's going to be a rough transition because we changed the meaning of this flag. How about in a follow-up we add validation for the publish options that checks if they are using `--tag` to specify the entire reference and error out explaining that the flags have changed?

We can identify  that `--tag` contains a reference by checking for `:`/`@`, neither of which are allowed in a docker tag.

Reply:
Good point.  I almost think it's warranted to delay this PR until another commit including this logic is pushed.  WDYT?  I can work on this.

Reply:
If you are up for it, that's great.

Reply:
Ok, I added logic to error accordingly for the tag option on publish.

Do we want to do something similar for the other commands that previously also had a `tag` opt (courtesy BundlePullOptions)?  As it stands, users who try to invoke `porter install --tag mybuns:v0.1.0` will just see a standard `Error: unknown flag: --tag` after this is merged (and need to check the docs to see `--reference` is the new opt).  Naturally, in the release notes, we can note this breaking cli opt change... not sure if we should also bake in a deprecation warning or error.

Reply:
Yeah, we could add back a Tag field and flag definitions and then check if it was populated and give a better error message. It would just be temporary until 1.0 when hopefully we stop making breaking changes for a bit.

Reply:
Ok, added it back as a deprecated flag (in Cobra terms this means it'll hide the flag by default and produce a deprecation warning when a value is provided).  When provided, we just assign reference to the value instead of explicitly erroring out -- WDYT?